hq_superset/api.py

```diff
@@ -0,0 +1,47 @@
+import json
+from http import HTTPStatus
+
+from flask import request
+from flask_appbuilder.api import BaseApi
+from flask_appbuilder.baseviews import expose
+from superset.superset_typing import FlaskResponse
+from superset.views.base import (
+    handle_api_exception,
+    json_error_response,
+    json_success,
+)
+
+from .models import DataSetChange
+
+
+class DataSetChangeAPI(BaseApi):
+    """
+    Accepts changes to datasets from CommCare HQ data forwarding
+    """
+
+    MAX_REQUEST_LENGTH = 10 * 1024 * 1024  # reject JSON requests > 10MB
+
+    def __init__(self):
+        self.route_base = '/hq_webhook'
```

Comment:
nit: do we want to keep a better url for this? I assume this is getting used internally and I don't fully understand where all this will be used. "webhook" seems quite open ended on what that could mean.

Reply:
Good point. "hq_datasource"? "hq_dataset"?

Reply:
how about just `dataset`? I see we have been following a convention to put a `hq_` in all names.
Is it to avoid conflict with superset routes?

Reply:
having `hq_` in all names makes me think that its coming from hq but hq does not have anything called dataset.
hence you also suggested `hq_datasource` but this view is changing dataset.

Reply:
> Is it to avoid conflict with superset routes?

Yes, exactly. Superset already has "dataset" routes.

> but this view is changing dataset.

Yes. ... Although it is the endpoint for DataSourceRepeaters. The URL for their ConnectionSettings will be 'hq_datasource/change". So I guess at least it makes sense in the context of the client/HQ.

Reply:
> Although it is the endpoint for DataSourceRepeaters

I am thinking that we should name things in here independent of what they are used for in CommCareHQ and more around what they mean here. I guess I will have a better suggestion of what this should be named if I look into this more deeply. Fine to leave it as it is then.
We are in the early phases of this product so would be nice to focus on good names early on.

Reply:
f0276dd